cubedash/_stac.py

```diff
@@ -266,7 +266,31 @@ def as_stac_item(dataset: DatasetItem) -> pystac.Item:
     return item
 
 
+def _provider_from_dict(d: dict) -> pystac.Provider:
```

Comment:
If we're going to add validation logic to this, I think it could stand to be a little more robust:
* We should handle the case of `"name" not in d` more gracefully than `Provider.from_dict` does
* If `roles` has been provided as a string, we should inform/warn that the expected format is as a list
* We should warn if `roles` contains non-string entries (including `None`)

Alternatively, some or all of this logic could reside in the method we will add to retrieve these values in the first place. This isn't a public function, so we could ensure that we have a valid provider definition at an earlier stage.